src/App.tsx

```diff
@@ -1,61 +1,115 @@
-import './App.scss';
+import React, { useState } from "react";
+import "./App.scss";
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from "./api/users";
+import todosFromServer from "./api/todos";
+
+import Users from './types/Users';
+import TodosWithUsers from "./types/TodosWithUsers";
+
+import { TodoList } from "./components/TodoList";
 
 export const App = () => {
+  const [users] = useState<Users[]>(usersFromServer);
+  const [todos, setTodos] = useState<TodosWithUsers[]>(todosFromServer.map(todo => ({
+    ...todo,
+    user: users.find(user => user.id === todo.userId) || null
+  })))
+  const [title, setTitle] = useState<string>("");
+  const [userId, setUserId] = useState<number | undefined>(undefined);
+  const [titleError, setTitleError] = useState<boolean>(false);
+  const [userError, setUserError] = useState<boolean>(false);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title || !userId) {
+      return;
```

Comment:
4) should show all errors if form is empty

this one you will pass if implement an error handler.